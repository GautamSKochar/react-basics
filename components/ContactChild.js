import React from 'react';

class ContactChild extends React.Component {
    constructor() {
        super();
        this.state = {
            // pageMounted: true,
            opacity: 1.0
        };
        console.log('Constructor');
    }

    componentWillMount() {
        console.log('Component will mount');
    };
    componentDidMount() {
        console.log('Component did mount');
        this.timer = setInterval(function () {
            var opacity = this.state.opacity;
            opacity = parseFloat(opacity) - .05;
            if (opacity < 0.1) {
                opacity = 1.0;
            }
            this.setState({
                opacity: opacity
            });
        }.bind(this), 100);
    };
    componentWillReceiveProps(nextProps) {
        console.log('Component will receive props', nextProps);
    };
    shouldComponentUpdate(nextProps, nextState) {
        console.log('Should Component update', nextProps, nextState);
        return true;
    }
    componentWillUpdate(nextProps, nextState) {
        console.log('Component will update', nextProps, nextState);
    }
    componentDidUpdate(prevProps, prevState) {
        console.log('Component Did Update', prevProps, prevState);
    }

    componentWillUnmount() {
        console.log("Component Will Unmount");
    }

    render() {
        
        return (
            
            <div className="well" style={{marginLeft: '20px', }}>
                <h4>Child Component</h4><br/>
                <h5>Component Lifecycle Examples</h5>
                <section>Some dummy text...................</section>
                <div style={{ opacity: this.state.opacity }}>
                    <h5>Hello React...! Check Console for Component life cycle methods.</h5>
                </div>                
            </div>
        )
    }
}

export default ContactChild;