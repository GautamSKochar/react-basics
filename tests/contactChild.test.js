import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import ContactChild from '../components/ContactChild';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const raf = global.requestAnimationFrame = (cb) => {
    setTimeout(cb, 0)
}
configure({ adapter: new Adapter() });

test('ContactChild Component Should render as expected', () => {
    const component = shallow(<ContactChild />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
});