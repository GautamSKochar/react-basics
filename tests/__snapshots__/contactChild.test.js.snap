// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactChild Component Should render as expected 1`] = `
<div
  className="well"
  style={
    Object {
      "marginLeft": "20px",
    }
  }
>
  <h4>
    Child Component
  </h4>
  <br />
  <h5>
    Component Lifecycle Examples
  </h5>
  <section>
    Some dummy text...................
  </section>
  <div
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <h5>
      Hello React...! Check Console for Component life cycle methods.
    </h5>
  </div>
</div>
`;
